brute-force-core/src/main/java/com/bakdata/kafka/AbstractBruteForceConfig.java

```diff
@@ -0,0 +1,102 @@
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.utils.Utils;
+
+/**
+ * Base configuration for kafka-brute-force projects.
+ *
+ * <p>
+ * This allows users to:
+ *  <ul>
+ *      <li>toggle support for large message</li>
+ *      <li>toggle if no matches should lead to an exception</li>
+ *  </ul>
+ */
+public class AbstractBruteForceConfig extends AbstractConfig {
+    public static final String PREFIX = "brute.force.";
+
+    public static final String LARGE_MESSAGE_ENABLED_CONFIG = PREFIX + "large.message.enabled";
+    public static final String LARGE_MESSAGE_ENABLED_DOC = "Flag for enabling support for large-message-serde.";
+    public static final boolean LARGE_MESSAGE_ENABLED_DEFAULT = true;
+
+    public static final String IGNORE_NO_MATCH_CONFIG = PREFIX + "ignore.no.match";
+    public static final String IGNORE_NO_MATCH_DOC =
+            "If set, the deserialization won't fail and instead keep the data as a byte array. "
+                    + "This is equivalent to including ByteArray conversion in the corresponding conversion list.";
+    public static final boolean IGNORE_NO_MATCH_DEFAULT = true;
+
+    protected AbstractBruteForceConfig(final ConfigDef definition, final Map<?, ?> originals) {
+        super(definition, originals);
+    }
+
+    protected static ConfigDef baseConfigDef() {
+        return new ConfigDef()
+                .define(LARGE_MESSAGE_ENABLED_CONFIG, Type.BOOLEAN, LARGE_MESSAGE_ENABLED_DEFAULT, Importance.LOW,
+                        LARGE_MESSAGE_ENABLED_DOC)
+                .define(IGNORE_NO_MATCH_CONFIG, Type.BOOLEAN, IGNORE_NO_MATCH_DEFAULT, Importance.LOW,
+                        IGNORE_NO_MATCH_DOC);
+    }
+
+    public boolean largeMessageEnabled() {
+        return this.getBoolean(LARGE_MESSAGE_ENABLED_CONFIG);
+    }
+
+    public boolean ignoreNoMatch() {
+        return this.getBoolean(IGNORE_NO_MATCH_CONFIG);
+    }
+
+    /**
+     * Get a list of instances as configured by the given config.
+     *
+     * <p>
+     * Compared to {@link #getConfiguredInstances(String, Class)}, this method doesn't call
+     * {@link org.apache.kafka.common.Configurable#configure(Map)}. This allows to defer the configuration.
+     *
+     * @param config key for the list of classes
+     * @param baseClazz base class for all instances
+     * @return a list with instances of the base class
+     */
+    protected <T> List<T> getInstances(final String config, final Class<T> baseClazz) {
+        return this.getList(config).stream().map(configuredClazz -> {
+                    try {
+                        return (Class<? extends T>) Utils.loadClass(configuredClazz, baseClazz);
+                    } catch (final ClassNotFoundException e) {
+                        throw new KafkaException("Could not load class " + configuredClazz);
```

Comment:
Isn't KafkaException a bit too broad?

Reply:
Yes, I'm also not too happy with it, but it's based on [`AbstractConfig#getConfiguredInstance(Object, Class<T>, Map<String,Object>`)](https://github.com/apache/kafka/blob/6b93652a5469b66419189336a6a96d5742a19043/clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java#L389) and there they use KafkaException as well.